src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,229 @@
+import React, {
+  Children,
+  ReactElement,
+  useEffect,
+  useRef,
+  useState,
+} from 'react'
+import classnames from 'classnames'
+interface TooltipProps {
+  label: string
+  position?: 'top' | 'bottom' | 'left' | 'right'
+  className?: string
+  children: ReactElement
+}
+
+const TRIANGLE_SIZE = 5
+const SPACER = 2
+const tooltipID = `tooltip-${Math.floor(Math.random() * 900000) + 100000}`
+
+function isElementInViewport(
+  el: HTMLElement,
+  win = window,
+  docEl = document.documentElement
+): boolean {
+  const rect = el.getBoundingClientRect()
+  return (
+    rect.top >= 0 &&
+    rect.left >= 0 &&
+    rect.bottom <= (win.innerHeight || docEl.clientHeight) &&
+    rect.right <= (win.innerWidth || docEl.clientWidth)
+  )
+}
+
+export const Tooltip = (
+  props: TooltipProps & JSX.IntrinsicElements['span']
```

Comment:
see my comment about changing the signature of this component, I think this should be `JSX.IntrinsicElements['button']`